src/Validator/UniqueUrlValidator.php

```diff
@@ -36,14 +38,21 @@ public function __construct(PageManagerInterface $manager)
 
     public function validate($value, Constraint $constraint): void
     {
-        if (!$value instanceof PageInterface) {
-            $this->context->addViolation('The page is not valid, expected a PageInterface');
-
+        if (!$constraint instanceof UniqueUrl) {
+            throw new UnexpectedTypeException($constraint, UniqueUrl::class);
+        }
+        // do not validate on null, use NotNull instead
+        if (null === $value) {
             return;
         }
+        if (!$value instanceof PageInterface) {
+            throw new UnexpectedValueException($value, PageInterface::class);
+        }
 
-        if (!$value->getSite() instanceof SiteInterface) {
-            $this->context->addViolation('The page is not linked to a Site');
+        if (null === $value->getSite()) {
+            $this->context->buildViolation('error.uniq_url.no_site')
```

Comment:
We need to add translations for this key:
In https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Resources/translations/validators.en.xliff and other files.

Reply:
I can try to add them, but im probably not good with translate them

Reply:
You can add the one you know (I can give you the french one).

For others, either we use google translate, either we do nothing and rely on contribution when someone need it ;)

Reply:
added English and German translation